src/main/java/org/dependencytrack/model/AffectedVersionAttribution.java

```diff
@@ -0,0 +1,100 @@
+package org.dependencytrack.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import javax.jdo.annotations.Column;
+import javax.jdo.annotations.IdGeneratorStrategy;
+import javax.jdo.annotations.Index;
+import javax.jdo.annotations.PersistenceCapable;
+import javax.jdo.annotations.Persistent;
+import javax.jdo.annotations.PrimaryKey;
+import javax.jdo.annotations.Unique;
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.util.Date;
+import java.util.UUID;
+
+/**
+ * Model class for tracking the attribution of vulnerable software version reporting.
+ */
+@PersistenceCapable
+@Index(name = "AFFECTEDVERSIONATTRIBUTION_IDX", members = {"vulnerableSoftware"})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class AffectedVersionAttribution implements Serializable {
+
+    private static final long serialVersionUID = -2609603709255246845L;
+
+    @PrimaryKey
+    @Persistent(valueStrategy = IdGeneratorStrategy.NATIVE)
+    @JsonIgnore
+    private long id;
+
+    @Persistent
+    @Column(name = "ATTRIBUTED_ON", allowsNull = "false")
+    @NotNull
+    private Date attributedOn;
+
+    @Persistent
+    @Column(name = "SOURCE", allowsNull = "false")
+    private Vulnerability.Source source;
+
+    @Persistent(defaultFetchGroup = "true")
+    @Column(name = "VULNERABLE_SOFTWARE", allowsNull = "false")
+    @NotNull
+    private VulnerableSoftware vulnerableSoftware;
```

Comment:
This relationship may cause infinite recursion errors when instances of either `VulnerableSoftware` or `AffectedVersionAttribution` are serialized. Adding a `@JsonIgnore` here should prevent this from happening.